plugins/circleci/src/state/models/buildWithSteps.ts

```diff
@@ -0,0 +1,80 @@
+import { Dispatch, iRootState } from '../store';
+import { GitType, BuildWithSteps } from 'circleci-api';
+import { CircleCIApi } from 'api';
+
+export type BuildState = {
+  builds: Record<number, BuildWithSteps>;
+  pollingIntervalId: number | null;
+  pollingState: PollingState;
+};
+
+const INTERVAL_AMOUNT = 1500;
+
+export enum PollingState {
+  Polling,
+  Idle,
+}
+export const buildWithSteps = {
```

Comment:
I think exposing these as hooks might be nicer? And maybe switching to `useReducer` instead of redux.

By switching to hooks I mean making this a custom hook and then calling `useDispatch` etc here

Reply:
We started with hooks, but since we need global state for the settings, it kinda made sense to move to something that handles both global state *and* effects in a good way, allowing to collocate. Maybe we could've done the same thing with the raw hooks only, but do you think it's valuable now to migrate back to hooks @Rugvip?

Reply:
@Rugvip thoughts?

Reply:
There's no entangled global state here right? As in this model isn't consumed by anyone, and only consumes settings. Same for builds, it doesn't consume anything else than settings and isn't consumed by this model?

I'm thinking a good way to refactor this would be to put settings is a plain react context, and probably just put the provider for that around a router like you had before. Eventually relying on react-router 6. Then both this and the builds model should be possible to refactor into one or more hooks?

I do think it's worth spending time on these plugins to make sure they are good candidates for copying into new plugins. Especially with this one being intended as the base CI plugin.

Reply:
> There's no entangled global state here right?
Seems to be right and I can't argue with the last statement too. I'll fix the errors logging together with the state management.

Reply:
Ok. That's where it's going https://github.com/Nek/backstage/pull/11/files